src/unitxt/inference.py

```diff
@@ -466,19 +469,48 @@ def verify(self):
         ), "Either 'model_name' or 'deployment_id' must be specified, but not both at the same time."
         super().verify()
 
-    def _infer(self, dataset):
+    def _load_model_and_params(self):
         from ibm_watsonx_ai.foundation_models import ModelInference
 
         model = ModelInference(
             model_id=self.model_name,
             deployment_id=self.deployment_id,
             api_client=self.client,
         )
+        params = self.to_dict([WMLInferenceEngineParamsMixin], keep_empty=False)
+
+        return model, params
+
+    def _infer(self, dataset):
+        model, params = self._load_model_and_params()
 
         return [
             model.generate_text(
                 prompt=instance["source"],
-                params=self.to_dict([WMLInferenceEngineParamsMixin], keep_empty=False),
+                params=params,
             )
             for instance in dataset
         ]
+
+    def _infer_log_probs(self, dataset):
+        model, params = self._load_model_and_params()
+
+        params = {
+            **params,
+            "return_options": {
```

Comment:
I din't think this needs to be hard coded.  It can be set in the params externally.

Reply:
It can, although one would need to specify this specific combination of parameters for this to work. Hard coding removes this burden from the user.

Reply:
Yes. But sometimes people want to include input tokens in the results and sometime not.  Also the number of top predictions may change .  I suggest that the code will fail with a clear error message, if it does not get what it expects.

Reply:
They do not really have the option - you must set either `input_tokens=True` or `input_text=True`, otherwise the client returns an error. And if you set `input_text=True,` the input is included in the "generated_text" field, which can cause confusion and inconsistent behavior. So given that this is the infer_with_logprobs method, and this is the only real way to get logprobs, I still think it makes more sense to pass these parameters than expect the user to figure out for themselves that it is the way to do this. The only real degree of freedom here is the top_n tokens, where the maximum one can request is 5, but if you really want you can ask to get less.

Reply:
I understand. Some points:
1. I would think it would be possible to possible to return only the generated token (which is what you need)

![image](https://github.com/user-attachments/assets/6d34854b-8982-4a74-89b0-2183e620d9d5)

If not, it's a bug.

If we select to  return the input_tokens, then it will include the "Yes"/"No" tokens of the prompt.  Is that why you had the search for end of the token list?  
 
1 . we need the behavior to be consistent across all inference engines (OpenAI ).  So if one inference engine sets the default params, the others should too.  Note that people may want to use the log probabilities in different ways than yours.

2. If we s set parameters We need to check if the user set these parameters already, an if so raise an error / warning.

Reply:
1. Indeed looks like a bug. I saw the API documentation but in practice the behavior is different. 
<img width="649" alt="image" src="https://github.com/user-attachments/assets/76582552-1470-41f4-bd1d-d246023762c1">
2. Certainly, we can raise an error if the user tries to pass the wrong parameters. But given that this specific API behaves in a very unpredictable and undocumented manner currently (unlike say OpenAI), I really think we should have a default behavior in this method.

Reply:
Regarding searching the end of the token list - the processor that searches from the end is for a situation where the answer is expected to come at the end (say, if the model is asked to generate an explanation before answering).

Reply:
done